eitprocessing/datahandling/eitdata.py

```diff
@@ -36,25 +33,27 @@ class is meant to hold data from (part of) a singular continuous measurement.
     Several convenience methods are supplied for calculating global impedance, calculating or removing baselines, etc.
     """  # TODO: fix docstring
 
-    path: Path | list[Path]
+    path: str | Path | list[Path | str] | list[Path] | list[str]
     nframes: int
     time: np.ndarray = field(repr=False)
     framerate: float
     vendor: Vendor
-    phases: list = field(default_factory=list, repr=False)
-    events: list = field(default_factory=list, repr=False)
-    label: str | None = None
+    label: str
     name: str | None = None
     pixel_impedance: np.ndarray = field(repr=False, kw_only=True)
 
     def __post_init__(self):
         if not self.label:
             self.label = f"{self.__class__.__name__}_{id(self)}"
+        if isinstance(self.path, str):
+            self.path = Path(self.path)
+        elif isinstance(self.path, list):
+            self.path = [Path(p) for p in self.path]
         self.name = self.name or self.label
         self._check_equivalence = ["label", "vendor", "framerate"]
 
     @staticmethod
-    def ensure_path_list(path: str | Path | list[str | Path]) -> list[Path]:
+    def ensure_path_list(path: str | Path | list[str | Path] | list[str] | list[Path]) -> list[Path]:
```

Comment:
I made this change because type checkers did not understand. I now fixed this by creating two overloads in b87f829d28216ea199025fc25e91c093ff093fad.

Reply:
why not always output a list of paths, even if it has a length of 1? Is there any implementation where this would be problematic? It would definitely make the code more readable to avoid the overload methods here.